src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,58 @@
+import "@testing-library/jest-dom/extend-expect"
+import React, { useState } from "react"
+import { act, fireEvent, render } from "@testing-library/react"
+
+import Checkbox, { ICheckboxProps } from "./Checkbox"
+
+const CheckboxTemplate = (props: ICheckboxProps) => {
+  const [checked, setChecked] = useState(false)
+
+  return (
+    <Checkbox
+      checked={checked}
+      onChange={() => setChecked((previous) => !previous)}
+      {...props}
+    />
+  )
+}
+
+describe("Checkbox", () => {
+  it("can be checked", () => {
+    const mockedFunction = jest.fn()
+    const { getByTestId } = render(
+      <CheckboxTemplate onChange={mockedFunction} />
+    )
+
+    const checkbox = getByTestId("checkbox")
+
+    expect(checkbox).toBeInTheDocument()
+
+    act(() => {
+      fireEvent.click(checkbox)
+    })
```

Comment:
There's no need to wrap fireEvent in act in this case, as there's nothing assync going on in the component and afaik fireEvent is already wrapped in act under the hood

```suggestion
    fireEvent.click(checkbox)
```